spreadsheet/formula/evaluator.go

```diff
@@ -19,14 +19,20 @@ import (
 // the results.
 type Evaluator interface {
 	Eval(ctx Context, formula string) Result
+	SetCache(key string, value Result)
```

Comment:
Same comment regarding interface, thinking about semantic versioning compatibility.  We can keep it if needed.  Some risk of breaking compatibility if someone is relying on the Evaluator interface as defined

Reply:
I thought about it and that's why I made this https://github.com/unidoc/unioffice/pull/366/files#diff-1fa3e06aaf304a444f44671cfdba3f93. Now if someone wants to make Evaluator without caching he can embed noCache like this:

type myEval struct {
        noCache
}

without write his own methods to implement the interface

If it is ok, I will do something similar to LastEvalIsRef method

Reply:
OK.  Can you add that to the documentation above?  Something like
// NOTE: ... above the interface

Reply:
done